CMakeLists.txt

```diff
@@ -92,20 +93,20 @@ endif()
 
 if(ANDROID)
     # Optimize for size vs. performance for Android. The has the following implications:
-    # - Replace `-O2` with `-Oz`.
-    # - Clang LTO does not support `-Oz` so keep using `-O2` for that.
     # - Add `-ffunction-sections` and `-fdata-sections`. This requires that `-Wl,-gc-sections` are used when creating
     #   the final .so file.
     # - `-fstrict-aliasing` is inherited from NDK r10e.
     # - `-fomit-frame-pointer` is inherited from NDK r10e.
-    #
-    # On some architectures char is unsigned by default. Make it signed
-    if(CMAKE_INTERPROCEDURAL_OPTIMIZATION)
-        set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2")
-    else()
-        set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Oz")
-    endif()
-    add_compile_options(-fdata-sections -ffunction-sections -fomit-frame-pointer -fsigned-char -fstrict-aliasing -funwind-tables -no-canonical-prefixes)
+    # - On some architectures char is unsigned by default. Make it signed
+    # - Compile with -Oz in Release because on Android we favor code size over performance
+    # 
+    add_compile_options(-fdata-sections -ffunction-sections -fomit-frame-pointer -fsigned-char -fstrict-aliasing -funwind-tables -no-canonical-prefixes $<$<CONFIG:Release>:-Oz>)
+
+    # CMake hard-codes the `-fuse-ld=gold` linker flag on Android for Clang when LTO is enabled, but we want to use lld instead.
+    # Unset for now, until it's fixed upstream.
+    # see https://github.com/Kitware/CMake/blob/af695cee7346fa0eee478f71952fd02f0df989cf/Modules/Compiler/Clang.cmake#L81-L84
+    unset(CMAKE_C_LINK_OPTIONS_IPO)
```

Comment:
Suggest to use `STRING(REGEX REPLACE "-fuse-ld=gold" "" CMAKE_CXX_LINK_OPTIONS_IPO ${CMAKE_CXX_LINK_OPTIONS_IPO})` instead to avoid clearing other flags in the future?

Reply:
This is fixed in the upcoming CMake 3.20. I'll remove the workaround and merge once we've updated.